lightning/src/routing/router.rs

```diff
@@ -1119,43 +1119,83 @@ const _GRAPH_NODE_SMALL: usize = 64 - core::mem::size_of::<RouteGraphNode>();
 #[cfg(any(ldk_bench, not(any(test, fuzzing))))]
 const _GRAPH_NODE_FIXED_SIZE: usize = core::mem::size_of::<RouteGraphNode>() - 64;
 
+/// A [`CandidateRouteHop::FirstHop`] entry.
+#[derive(Clone, Debug)]
+pub struct CandidateFirstHop<'a> {
```

Comment:
Could you name these such that the attributive noun (`FirstHop`) modifying the primary noun (`Candidate`) comes first? (e.g., `FirstHopCandidate`)